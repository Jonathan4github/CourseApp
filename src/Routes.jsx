import React, { Fragment } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import Header from './components/common/Header';
import HomePage from './components/home/HomePage';
import CoursesPage from './components/course/CoursesPage';
import AuthorsPage from './components/author/AuthorsPage';
import AboutPage from './components/about/AboutPage';
import ManageCoursePage from './components/course/ManageCoursePage';
import ManageAuthorPage from './components/author/ManageAuthorPage';
import NoPageFound from './components/common/NotFoundPage';


const Routes = () => (
  <BrowserRouter>
    <Fragment>
      <div className="container-fluid">
        <Header />
      </div>
      <div className="container main-content">
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route path="/about" component={AboutPage} />
          <Route exact path="/courses/:id" component={ManageCoursePage} />
          <Route exact path="/courses/add" component={ManageCoursePage} />
          <Route exact path="/courses" component={CoursesPage} />
          <Route exact path="/authors/:id" component={ManageAuthorPage} />
          <Route exact path="/authors/add" component={ManageAuthorPage} />
          <Route exact path="/authors" component={AuthorsPage} />
          <Route component={NoPageFound} />
        </Switch>
      </div>
    </Fragment>
  </BrowserRouter>
);

export default Routes;
