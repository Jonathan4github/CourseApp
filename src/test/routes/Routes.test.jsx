/* eslint-disable */
import Routes from '../../Routes';


describe('<Routes />', () => {
  it('renders entire app route components without crashing', () => {
    const shallowWrapper = shallow(<Routes />);
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
