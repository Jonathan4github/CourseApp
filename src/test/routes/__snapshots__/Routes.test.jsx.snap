// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> renders entire app route components without crashing 1`] = `
<BrowserRouter>
  <div
    className="container-fluid"
  >
    <Connect(Header) />
  </div>
  <div
    className="container main-content"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/courses/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/courses/add"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/courses"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/authors/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/authors/add"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/authors"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</BrowserRouter>
`;
